crates/node-cli/src/main.rs

```diff
@@ -241,9 +241,9 @@ async fn run(args: Args) -> anyhow::Result<()> {
     tokio::select! {
         _ = api => {},
         _ = ctrl_c => {},
-        r = node_future => {
-            if let Err(e) = r {
-                #[cfg(feature = "tracing")]
+        _r = node_future => {
+            #[cfg(feature = "tracing")]
+            if let Err(e) = _r {
                 tracing::error!("Critical error on relayer, state derivation or validation streams: {e}")
             }
```

Comment:
If feature is disabled, does this if block just get skipped?

Reply:
Yeah exactly :ok_hand:  I noticed we were getting an unused binding warning when tracing was disabled in the CLI, so I changed this to feature gate the whole block instead of just the log itself